src/commands/crashlytics-mappingfile-generateid.ts

```diff
@@ -0,0 +1,45 @@
+import { Command } from "../command";
+import * as utils from "../utils";
+
+import { fetchBuildtoolsJar, runBuildtoolsCommand } from "../crashlytics/buildToolsJarHelper";
+import { Options } from "../options";
+import { FirebaseError } from "../error";
+
+interface CommandOptions extends Options {
+  resourceFile: string;
+}
+
+interface JarOptions {
+  resourceFilePath: string;
+}
+
+export const command = new Command("crashlytics:mappingfile:generateid")
+  .description(
+    "generate a mapping file id and write it to an Android resource file, which will be built into the app"
+  )
+  .option(
+    "--resource-file <resourceFile>",
+    "Path to the Android resource XML file that will be created or updated."
+  )
+  .option("--debug", "print debug output and logging")
+  .action(async (options: CommandOptions) => {
+    const debug = !!options.debug;
+    // Input errors will be caught in the buildtools jar.
+    const resourceFilePath = options.resourceFile;
+    if (!resourceFilePath) {
+      throw new FirebaseError(
+        "set --resource-file <resourceFile> to an Android resource file path, e.g. app/src/main/res/values/crashlytics.xml"
+      );
+    }
+    const jarFile = await fetchBuildtoolsJar();
+    const jarOptions: JarOptions = { resourceFilePath };
+
+    utils.logBullet(`Updating resource file: ${resourceFilePath}`);
+    const uploadArgs = buildArgs(jarOptions);
+    runBuildtoolsCommand(jarFile, uploadArgs, debug);
+    utils.logBullet("Successfully updated mapping file id");
+  });
```

Comment:
[optional] it looks like to me that this returns `undefined`. Would it be possible to return something useful for machine output (i.e. the command run with `--json`)? 

Thinking about it a bit myself - if it's going to the JAR and doesn't really have a good way of parsing it out, might be more tricky than it's worth. `--json` output isn't required, but it does make it nicer to deal with if the CLI is run in a script

Reply:
Bryan & I discussed in person and decided to keep the return `undefined` for now, largely because parsing the output of the executable jar is going to be brittle. We'll revisit if we get feedback from customers using the `--json` switch.